chips/cc26xx/src/gpio.rs

```diff
@@ -0,0 +1,190 @@
+use core::cell::Cell;
+use core::ops::{Index, IndexMut};
+use ioc;
+use kernel::common::VolatileCell;
+use kernel::hil;
+
+const NUM_PINS: usize = 32;
+
+#[repr(C)]
+pub struct GpioRegisters {
+    _reserved0: [u8; 0x90],
+    pub dout_set: VolatileCell<u32>,
+    _reserved1: [u8; 0xC],
+    pub dout_clr: VolatileCell<u32>,
+    _reserved2: [u8; 0xC],
+    pub dout_tgl: VolatileCell<u32>,
+    _reserved3: [u8; 0xC],
+    pub din: VolatileCell<u32>,
+    _reserved4: [u8; 0xC],
+    pub doe: VolatileCell<u32>,
+    _reserved5: [u8; 0xC],
+    pub evflags: VolatileCell<u32>,
+}
+
+pub const GPIO_BASE: *mut GpioRegisters = 0x4002_2000 as *mut GpioRegisters;
+
+pub struct GPIOPin {
+    regs: *mut GpioRegisters,
+    pin: usize,
+    pin_mask: u32,
+    client_data: Cell<usize>,
+    client: Cell<Option<&'static hil::gpio::Client>>,
+}
+
+impl GPIOPin {
+    const fn new(pin: usize) -> GPIOPin {
+        GPIOPin {
+            regs: GPIO_BASE,
+            pin: pin,
+            pin_mask: 1 << (pin % NUM_PINS),
+            client_data: Cell::new(0),
+            client: Cell::new(None),
+        }
+    }
+
```

Comment:
Same here,
```rustc
impl GPIOPin {
     const fn new(pin: usize) -> GPIOPin {
         GPIOPin {
              regs: 0x4002_2000 as *const GpioRegisters,
              // other code
}
```

Reply:
I think it is better to keep it in this case, since it is referenced at [L135](https://github.com/helena-project/tock/pull/747/files/016fd10e7329e5a283962d830c0b88de1e75a956#diff-4aaab6316a62ef81d22afe6ca8c3c7f9R136) as well

Reply:
Probably, missed that you had two different structs